src/decision/src/LidarDecision.cpp

```diff
@@ -31,6 +32,143 @@ void LidarDecision::scanCallBack(const sensor_msgs::LaserScan::ConstPtr& raw_sca
     // Deal with new messages here
 }
 
+
+
+
+
+double pi = 3.1415926;
+
+
+bool LidarDecision::obstacle_in_range(double min_angle, double max_angle, float obstacle_distance,
+                                      const sensor_msgs::LaserScan::ConstPtr& raw_scan) {
+    int i;
+    int init_pos_vec, end_pos_vec;
+
+    init_pos_vec = static_cast<int>((min_angle - raw_scan->angle_min) / raw_scan->angle_increment);
+    end_pos_vec = static_cast<int>((max_angle - raw_scan->angle_min) / raw_scan->angle_increment);
+
+    /*bool exist_obstacle = false;
+    for (i = init_pos_vec; i <= end_pos_vec; i++) {
+        if (raw_scan->ranges[i] < obstacle_distance) exist_obstacle = true;
+    }
+    if (exist_obstacle) return true;
+    else return false;*/
+    return std::any_of(
+             raw_scan->ranges.begin()+init_pos_vec, raw_scan->ranges.begin()+end_pos_vec,
+             [&](auto const& v){
+                 return v < obstacle_distance;
+             });
+
+}
+
+//the twist message to be sent out will be modified throught the following functions
```

Comment:
Any comments describing what any functions are doing should be in the header file in Javadoc format (see the README). If someone were to move some functions around later, and not notice this comment, it could confuse people further down the road.